src/state.py

```diff
@@ -219,31 +223,7 @@ def from_integration_data(cls, data: ops.RelationDataContent) -> "LocationConfig
             raise ConfigurationError(f"Config error: {err_msg}") from err
 
 
-def validate_path_value(value: str) -> str:
-    """Validate the value as a path.
-
-    Validation performed:
-    - The path is only consist of allowed characters.
-
-    Args:
-        value: The value to validate.
-
-    Raises:
-        ValueError: The validation failed.
-
-    Returns:
-        The value after validation.
-    """
-    # This are the valid characters for path in addition to `/`:
-    # a-z A-Z 0-9 . - _ ~ ! $ & ' ( ) * + , ; = : @
-    # https://datatracker.ietf.org/doc/html/rfc3986#section-3.3
-    valid_path = re.compile(r"[/\w.\-~!$&'()*+,;=:@]+", re.IGNORECASE)
-    if valid_path.fullmatch(value) is None:
-        raise ValueError("Path contains non-allowed character")
-    return value
-
-
-def check_nginx_time_str(time_str: str) -> None:
+def _check_nginx_time_str(time_str: str) -> None:
     """Check if nginx time str is valid.
```

Comment:
I think we should put the rules that makes time str valid in docstring. I thought you already did something like this. Was it in the other repo?